### PR TITLE
many: remove deadcode

### PR DESCRIPTION
**Pull request by mvo5**
I ran deadcode (remyoudompheng/go-misc/deadcode) over our codebase
and removed all the low hanging fruits.

### CODE REVIEW

File: overlord/ifacestate/ifacestate.go [hunk 1]
```diff
@@ -36,11 +36,6 @@ import (
 	"github.com/snapcore/snapd/snap"
 )
 
-var noConflictOnConnectTasks = func(task *state.Task) bool {
```
Comment (mvo5) [inline overlord/ifacestate/ifacestate.go]: @stolowski should double check this one, maybe a in-flight PR uses it?

**Comment by mvo5** [comment]
@bboozzoo Yeah, there is more to do :) Some are false positives, like bySlotRef which is used in tests. So maybe the right approach here is to move them into the tests. I did not look at snap-update-ns at all (mostly because I'm not sure if deadcode picks up the C<->go bridge properly).

File: overlord/ifacestate/ifacestate.go (hunk 1 above)
Comment (stolowski) [reply overlord/ifacestate/ifacestate.go]: It's fine, safe to remove, I forgot about it last time I touched conflict checking in ifacestate. Thanks!

Comment (pedronis) [reply overlord/ifacestate/ifacestate.go]: it was used by less streamlined/conservative conflict code, I made it unused in one of my refactorings but forgot to remove

**Pull request merged**